foundry/foundry.py

```diff
@@ -122,10 +126,30 @@ def load(self, name, download=True, globus=True, verbose=False, metadata=None, *
         else:
             res = metadata
 
-        # TODO: if object empty, handle
-        res = res[0]
-        res["dataset"] = res["projects"][self.config.metadata_key]
-        res["dataset"]["type"] = res["dataset"]["data_type"]
+        # we figured from our tests that at each of these three lines there could be an error
+        # if res is an empty list. (we also see that res will always be an empty list,
+        # never None or not a list)
+        try:
+            res = res[0] # if search returns multiple results, this automatically uses first result
+        except IndexError as e:
+            raise Exception("load: No metadata found for given dataset") from e
+        except BaseException as e:
+            raise Exception("load: An unknown error occurred") from e
+
+        try:
+            res["dataset"] = res["projects"][self.config.metadata_key]
```

Comment:
this could be an issue with "projects", or it could be an issue with the metadata_key. I would suggest changing the message here to something like `"not able to index with metadata key {}".format(self.config.metadata_key)`